diff --git a/vision/classification_and_detection/python/models/base_model_r34.py b/vision/classification_and_detection/python/models/base_model_r34.py
index ea224a7c..a79760d5 100644
--- a/vision/classification_and_detection/python/models/base_model_r34.py
+++ b/vision/classification_and_detection/python/models/base_model_r34.py
@@ -56,7 +56,7 @@ class ResNet18(nn.Module):
 class ResNet34(nn.Module):
     def __init__(self):
         super().__init__()
-        rn34 = resnet34(pretrained=True)
+        rn34 = resnet34(pretrained=None)
 
         # discard last Resnet block, avrpooling and classification FC
         self.layer1 = nn.Sequential(*list(rn34.children())[:6])
diff --git a/vision/classification_and_detection/python/models/ssd_r34.py b/vision/classification_and_detection/python/models/ssd_r34.py
index 63e596b4..0db97808 100644
--- a/vision/classification_and_detection/python/models/ssd_r34.py
+++ b/vision/classification_and_detection/python/models/ssd_r34.py
@@ -1,6 +1,6 @@
 import torch
 import torch.nn as nn
-from models.base_model_r34 import ResNet34
+from vision.classification_and_detection.python.models.base_model_r34 import ResNet34
 import numpy as np
 from math import sqrt, ceil
 import itertools
@@ -28,13 +28,13 @@ class Encoder(object):
         self.dboxes_xywh = dboxes(order="xywh").unsqueeze(dim=0)
         self.nboxes = self.dboxes.size(0)
         #print("# Bounding boxes: {}".format(self.nboxes))
-        self.scale_xy = torch.tensor(dboxes.scale_xy)
-        self.scale_wh = torch.tensor(dboxes.scale_wh)
+        self.scale_xy = torch.tensor(dboxes.scale_xy, dtype=torch.float32)
+        self.scale_wh = torch.tensor(dboxes.scale_wh, dtype=torch.float32)
     
     
     def decode_batch(self, bboxes_in, scores_in,  criteria = 0.45, max_output=200):
-        self.dboxes = self.dboxes.to(bboxes_in)
-        self.dboxes_xywh = self.dboxes_xywh.to(bboxes_in)
+        self.dboxes = self.dboxes.to(bboxes_in.dtype)
+        self.dboxes_xywh = self.dboxes_xywh.to(bboxes_in.dtype)
         bboxes, probs = scale_back_batch(bboxes_in, scores_in,self.scale_xy,self.scale_wh,self.dboxes_xywh)
         boxes = []; labels=[]; scores=[]
         for bbox, prob in zip(bboxes.split(1, 0), probs.split(1, 0)):
@@ -60,9 +60,11 @@ class Encoder(object):
             if i == 0: continue
             
             score = score.squeeze(1)
-            mask = score > 0.05
+            mask = (score > 0.05).float().unsqueeze(1)
+            bboxes = bboxes_in * mask
+            score = score * mask.squeeze(1)
 
-            bboxes, score = bboxes_in[mask, :], score[mask]
+            # bboxes, score = bboxes_in[mask, :], score[mask]
             if score.size(0) == 0: continue
 
             score_sorted, score_idx_sorted = score.sort(dim=0)
@@ -183,7 +185,7 @@ class DefaultBoxes(object):
                 for i, j in itertools.product(range(sfeat_w), range(sfeat_h)):
                     cx, cy = (j+0.5)/fkh[idx], (i+0.5)/fkw[idx]
                     self.default_boxes.append((cx, cy, w, h)) 
-        self.dboxes = torch.tensor(self.default_boxes)
+        self.dboxes = torch.tensor(self.default_boxes, dtype=torch.float32)
         self.dboxes.clamp_(min=0, max=1)
         # For IoU calculation
         self.dboxes_ltrb = self.dboxes.clone()
